client/me/profile-links-add-wordpress/index.jsx

```diff
@@ -157,84 +157,94 @@ export default React.createClass( {
+					{ translate( 'Please select one or more sites to add to your profile.' ) }
 				</p>
 				<ul className="profile-links-add-wordpress__list">
 					{ this.renderAddableSites() }
 				</ul>
 				{ this.possiblyRenderError() }
 				<FormButton
 					disabled={ ( 0 === checkedCount ) ? true : false }
-					onClick={ this.recordClickEvent( 'Add WordPress Sites Button' ) }
+					onClick={ recordClickEvent( 'Add WordPress Sites Button' ) }
 				>
-					{ this.translate( 'Add Site', 'Add Sites', { count: checkedCount } ) }
+					{ translate( 'Add Site', 'Add Sites', { count: checkedCount } ) }
 				</FormButton>
 				<FormButton
 					className="profile-links-add-wordpress__cancel"
 					isPrimary={ false }
-					onClick={ this.recordClickEvent( 'Cancel Add WordPress Sites Button', this.onCancel ) }
+					onClick={ recordClickEvent( 'Cancel Add WordPress Sites Button', this.onCancel ) }
 				>
-					{ this.translate( 'Cancel' ) }
+					{ translate( 'Cancel' ) }
 				</FormButton>
 			</form>
 		);
-	},
+	}
 
 	renderInvitationForm() {
+		const { translate } = this.props;
+
 		return (
 			<form>
 				<p>
 					{
-						this.translate(
+						translate(
 							'You have no public sites on WordPress.com yet, but if you like you ' +
-							'can create one now.  You may also add self-hosted WordPress ' +
+							'can create one now. You may also add self-hosted WordPress ' +
 							'sites here after installing {{jetpackLink}}Jetpack{{/jetpackLink}} on them.',
 							{
 								components: {
 									jetpackLink: <a
 													href="#"
 													className="profile-links-add-wordpress__jetpack-link"
-													onClick={ this.recordClickEvent( 'Jetpack Link in Profile Links', this.onJetpackMe ) }
+													onClick={ recordClickEvent( 'Jetpack Link in Profile Links', this.onJetpackMe ) }
 												/>
 								}
 							}
 						)
 					}
 				</p>
 				<FormButton
-					onClick={ this.recordClickEvent( 'Create Sites Button in Profile Links', this.onCreateSite ) }
+					onClick={ recordClickEvent( 'Create Sites Button in Profile Links', this.onCreateSite ) }
 					>
-					{ this.translate( 'Create Site' ) }
+					{ translate( 'Create Site' ) }
 				</FormButton>
 				<FormButton
 					className="profile-links-add-wordpress__cancel"
 					isPrimary={ false }
-					onClick={ this.recordClickEvent( 'Cancel Add WordPress Sites Button', this.onCancel ) }
+					onClick={ recordClickEvent( 'Cancel Add WordPress Sites Button', this.onCancel ) }
 				>
-					{ this.translate( 'Cancel' ) }
+					{ translate( 'Cancel' ) }
 				</FormButton>
 			</form>
 		);
-	},
+	}
 
 	render() {
 		return (
 			<div>
-				{ 0 === sites.getPublic().length ? this.renderInvitationForm() : this.renderAddableSitesForm() }
+				{ 0 === this.props.publicSites.length ? this.renderInvitationForm() : this.renderAddableSitesForm() }
 			</div>
 		);
 	}
-} );
+}
+
+export default connect(
+	( state ) => ( {
+		sites: keyBy( getSites( state ), 'ID' ),
```

Comment:
Agreed. There might be a way.

`this.props.sites` seems to only be used to obtain `name` and `URL` in preparation of the call to `this.props.userProfileLinks.addProfileLinks`. In that case, I would define an action creator:

```js
// pseudo-code-ish
const addProfileLinks = ( inputs ) => ( dispatch, getState ) => {
  const links = inputs
    .filter( isSiteInput )
    .map( toSiteId )
    .map( getSite( getState(), _ ) )
    .map( pick( _, [ 'name', 'URL' ] ) );

  if ( links.length ) userProfileLinks.addProfileLinks( links );
};
```

If not, at the very least you should cache this selection, since `keyBy` could potentially process a large number of sites and, more importantly, will always return a new reference.